api/v1alpha1/project_config_types.go

```diff
@@ -46,6 +53,30 @@ type PromotionPolicy struct {
 	AutoPromotionEnabled bool `json:"autoPromotionEnabled,omitempty" protobuf:"varint,2,opt,name=autoPromotionEnabled"`
 }
 
+// WebhookReceiverConfig is a resource type that describes the configuration
+// for a receiver.
+type WebhookReceiverConfig struct {
+	// Type is the type of the receiver.
+	//
+	// TODO: Add more receiver enum types(e.g. Dockerhub, Quay, Gitlab, etc...)
+	// +kubebuilder:validation:Enum=GitHub;
+	Type string `json:"type,omitempty" protobuf:"bytes,2,opt,name=type"`
+	// Secret is the name of the secret that contains the credentials for the
+	// receiver.
+	//
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
+	SecretRef string `json:"secretRef,omitempty" protobuf:"bytes,4,opt,name=secretRef"`
+}
+
+// WebhookReceiver is a resource type that describes a path used
```

Comment:
"Resource type" == CRD, so we should only use that phrase for things that are "top level" so to speak.